### PR TITLE
Fix Go dependency + Update Brew install Script

### LINKED ISSUE #35: On Mac, Cannot install Pocket via Homebrew due to outdated GO dependency
While installing the pocket CLI via homebrew on my mac dev laptop I realized i couldn't because Brew would throw errors that the `go@1.18` is not available upstream.  I worked around it initially by editing the code in my brew install to allow installing deprecated packages and was able to get it going and working, however this is obviously not ideal and i didn't want to do it again while setting up a new Mac.

When i checked this repo i found a handful of minor errors, none too serious, save for the very outdated Go dependency, which as expected from my output trying to install the pocket-cli set to `go@1.18` which has long since been deprecated by Homebrew.  I checked the current dependency in the Pocket-Core repo see that its now `go@1.21` and need to update it to match so that people can install the pocket-cli using homebrew without having to hack the brew install and make deprecated packages installable.

I also noticed that the readme file was out of date, its instructions/script to install homebrew on Mac was very outdated, from when Mac still used bash as the default shell.  It must be upgraded to the latest install script for mac from Homebrew's website, since the newer (5 years old at least now likely longer) Macs changed to using zsh as their default shell so the bash command has to be used and theres a new link for it to curl.  It no longer relies on an existing ruby install but rather just bash.  It also requires `sudo` for most installs.

Also theres a handful of minor formatting errors.

### PR DESCRIPTION
**Pull request by alexpeterson91**
fixes #35 

updated go dependency to `go@1.21` 

updated install script for homebrew on mac

Fixed random formatting errors

### CODE REVIEW

**Comment by alexpeterson91** [comment]
@Olshansk

File: README.md [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
-You should have brew install.
+You need to have the package manager Homebrew installed.
 
-To install Homebrew on Mac:
+To install Homebrew on Mac, run the following command in a terminal:
 
-```bash
-/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+```zsh
+sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
Comment (Olshansk) [inline README.md]: Why does this need to be sudo?

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (alexpeterson91) [reply README.md]: on most macs this is the output without sudo, 
```
% /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" 
==> Checking for `sudo` access (which may request your password)...
Password:
```

So i suppose it's not needed in the command but you need the password whether or not you use sudo in the command.
```suggestion
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
I can remove it if you desire.

Comment (Olshansk) [reply README.md]: 1. Remove sudoe
2. Revert to bash. I also use zsh but keep it general purpose

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Comment by alexpeterson91** [comment]
@Olshansk got it, will remove sudo for sure, and i was gonna say why not leave the mac homebrew install command with the language defined as zsh because thats just the default shell for all macs and this is a mac command with the linux command separate.  
However i just checked and linuxbrew is deprecated, like 5 years now. was merged into regular homebrew and it's installed the same way so im gonna set the defined language to bash since thats whats needed for linux and mac, and it doesnt affect the way anything is rendered anyway at least in this case. 
Also i inspected [Brew.sh](https://brew.sh) 's home page with the script and it defines the language as bash as well. 
So im gonna make a couple other changes namely removing the linux command and move linux instructions to merge with macOS as they use the exact same command and the exact same source and language.
Let me know if you'd like me to include a link to the homebrew Mac flat package installer `.pkg` file.
and finally i changed the expected output from coinbase passphrase to keybase since thats the terminology used by pocket core, however when i run the start command it first initializes several things then says no keys in keybase, create using X command and exits. let me know if i should update the expected respnse

**Review by Olshansk — APPROVED** [review]

**Comment by Olshansk** [comment]
Please "squash & merge" when ready

**Comment by alexpeterson91** [comment]
> Please "squash & merge" when ready

i dont have write access to merge, must be repo rules

Screenshot 2024-05-02 at 3 44 45 PM

**Pull request merged**